package.json

```
@@ -8,7 +8,7 @@
     "build:dev:watch": "npm-run-all -p -l sass:watch webpack:dev:watch",
     "build:prd": "npm-run-all -p -l sass webpack:prd",
     "sass": "sass src/sass:src/dist/css",
-    "sass:watch": "npm run sass && sass src/sass:src/dist/css --watch --recursive",
+    "sass:watch": "sass --watch src/sass:src/dist/css src/Site/views/languageforge/theme/default/sass:src/dist/css src/angular-app:src/dist/css",
```

I am trying to figure out why the extra path mappings are needed.  My thinking is there shouldn't be a need for the second and third path mappings that you've included on the command line, since they are included as import statements in [`src/sass/languageforge.scss`](https://github.com/sillsdev/web-languageforge/blob/develop/src/sass/languageforge.scss#L1) 

First line in `languageforge.scss`:
```
@import "../Site/views/languageforge/theme/default/sass/languageforge_default.scss";
```
The other sass files in angular-app are explicitly imported one line at a time.

It seems fine to remove the `--recursive` since that doesn't actually seem necessary, as well as the redundant `npm run sass`.  Thanks for fixing that up.

```suggestion
    "sass:watch": "sass --watch src/sass:src/dist/css",
```

Shouldn't it work with the simplified suggestion above?  I haven't tested it.

This comment originally appeared over at https://github.com/sillsdev/web-languageforge/pull/1160#discussion_r712823597 until I realized it should actually be on this PR.  Sorry for the confusion.

---

> I am trying to figure out why the extra path mappings are needed. My thinking is there shouldn't be a need for the second and third path mappings that you've included on the command line, since they are included as import statements in src/sass/languageforge.scss

`sass` only watches changes in a directory (or a file)...not, "watch this file and any of its dependencies, i.e., anything it `@import`s as well".  See https://sass-lang.com/documentation/cli/dart-sass.  Additionally, `@import` is deprecated anyway...typically you'd see something like, `@use '.../some_partial_scss'`.  I didn't take the time to test whether having the newer syntax in there would've helped the watch work more simply but it's always something we could come back to if it's too bothersome in this current state.  Just keep in mind, migrating to`~@use` will require some additional changes to accommodate the namespacing that comes in with the newer syntax.

> It seems fine to remove the --recursive since that doesn't actually seem necessary
That option was available in the previous `node-sass` but is not an option for `sass`

---

Thank you for the explanation @longrunningprocess .  I didn't know about upgrading `@import` statements to `@use` statements.  It also makes sense that the sass watch needs to have different arguments than the vanilla compile step since the watch parameters are for what *files* it should watch, not which files are entry points for compilation.